pkg/generate/git/repository.go

```diff
@@ -205,27 +196,38 @@ func (r *repository) AddLocalConfig(location, name, value string) error {
 }
 
 // CloneWithOptions clones a remote git repository to a local directory
-func (r *repository) CloneWithOptions(location string, url string, opts CloneOptions) error {
-	args := []string{"clone"}
-	if opts.Quiet {
-		args = append(args, "--quiet")
-	}
-	if opts.Recursive {
-		args = append(args, "--recursive")
-	}
-	if opts.Shallow {
-		args = append(args, "--depth=1")
-		r.shallow = true
+func (r *repository) CloneWithOptions(location string, url string, args ...string) error {
+	gitArgs := []string{"clone"}
+	gitArgs = append(gitArgs, args...)
+	gitArgs = append(gitArgs, url)
+	gitArgs = append(gitArgs, location)
+
+	// We need to check to see if we're importing reference information, for
+	// for error checking later on
+	for _, opt := range gitArgs {
+		if opt == "--depth=1" {
```

Comment:
I'm unsure of what you mean, should I add a "shallowGitArgs[]" array at the top?